### ISSUE TITLE
[BUG] Basic HTTPS authentication not working for webkit browser in windows operating system

### ISSUE THREAD

**Issue by nanipunepalle** [microsoft/playwright#28912]
### System info
- Playwright Version: v1.40.1
- Operating System: Windows 10
- Browser: WebKit
- Other info:

### Source code

example.spec.ts
```
import { test, expect, chromium, webkit } from "@playwright/test";

test("User", async ({ browser, isMobile }) => {

    const context = await browser.newContext({
        httpCredentials: {
            username: process.env.USERNAME ?? "",
            password: process.env.PASSWORD ?? "",
        }
    });

    const page = await context.newPage();
    await page.goto("####URL####"); 
    !isMobile && await expect(page.locator('[data-text="test"]')).toBeVisible();
});
```

playwright.config.ts
```

import { defineConfig, devices } from "@playwright/test";

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
require("dotenv").config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
    testDir: "./tests",
    /* Run tests in files in parallel */
    fullyParallel: true,
    /* Fail the build on CI if you accidentally left test.only in the source code. */
    forbidOnly: !!process.env.CI,
    /* Retry on CI only */
    retries: process.env.CI ? 2 : 0,
    /* Opt out of parallel tests on CI. */
    workers: process.env.CI ? 1 : undefined,
    /* Reporter to use. See https://playwright.dev/docs/test-reporters */
    reporter: "html",
    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
    use: {
        /* Base URL to use in actions like `await page.goto('/')`. */
        // baseURL: 'http://127.0.0.1:3000',

        /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
        trace: "on-first-retry",
    },

    /* Configure projects for major browsers */
    projects: [
        {
            name: "chromium",
            use: {
                ...devices["Desktop Chrome"],
                launchOptions: {
                    args: [
                        '--auth-server-allowlist="_"',
                    ],
                },
            },
        },
        {
            name: "webkit",
            use: {
                ...devices["Desktop Safari"],
                // headless: false
            },
        },
    ],

    /* Run your local dev server before starting the tests */
    // webServer: {
    //   command: 'npm run start',
    //   url: 'http://127.0.0.1:3000',
    //   reuseExistingServer: !process.env.CI,
    // },
});


```

**Steps**
- Use windows environment to perform tests
- Pick any public site that has basic https authentication and update goto site url with it
- Set the username and password env variables
- Run the test

**Expected**

The basic https authentication should work in webkit browser and chromium browser

**Actual**

The https authentication works well chrome browser but this doesn't work in webkit browser

**Note:** Both browsers(chromium and webkit) work well in ubuntu environment. This happens only in windows environment

**Comment by mxschmitt** [comment]
We have [various basic auth tests](https://github.com/microsoft/playwright/blob/main/tests/library/browsercontext-credentials.spec.ts) in place which pass on WebKit Windows, I also just tested it on a https server. Maybe your server is behaving different on your Windows machine?

Would it be possible to share a minimal reproducible of the server?

**Comment by nanipunepalle** [comment]
If you mean the backend authentication server, I cannot give reproducible but the backend server is .net IIS windows server. Kerberos authentication mechanism is used. Attaching network request and reponse of chrome and webkit

```
Webkit Request
URL
https://example.com/api/authenticate
Request Headers
Origin: https://exampledomain.com
Accept: */*
Referer: https://exampledomain.com/
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-site
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.4 Safari/605.1.15
Accept-Language: en-US

No response is received
```

```
Chrome request
URL
https://example.com/api/authenticate
Request Headers
Accept: */*
Accept-Language: en-US
Origin: https://exampledomain.com
Referer: https://exampledomain.com/
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.6099.28 Safari/537.36

response is successful
```

**Comment by dgozman** [comment]
@nanipunepalle Unfortunately, we won't be able to help without a repro that we can run locally. Perhaps your server triggers some unexpected code path in WebKit that our tests do not catch? It's impossible to tell without debugging.

> Also could you provide reference to flags that can be added to launch options of webkit browser

There is no reference for this. I think WebKit only supports a few Playwright-specific launch arguments that you should not touch anyway.

**Comment by dgozman** [comment]
Closing because we lack the information to act on this issue. If you can provide a full repro, please file a new issue by filling the "Bug Report" template, and link to this one.
